tests/Nelmio/Alice/ReferenceRangeNameTest.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of the Alice package.
+ *
+ * (c) Nelmio <hello@nelm.io>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Nelmio\Alice;
+
+class ReferenceRangeNameTest extends \PHPUnit_Framework_TestCase
+{
+    const USER = 'Nelmio\Alice\support\models\User';
+    const GROUP = 'Nelmio\Alice\support\models\Group';
+    const CONTACT = 'Nelmio\Alice\support\models\Contact';
+
+    /**
+     * @test
+     * @expectedException \UnexpectedValueException
+     */
+    public function loadFixturesByReferenceNotFound()
+    {
+        $managerMock = $this->getDoctrineManagerMock(null);
+
+        $files = [
+            __DIR__ . '/support/fixtures/reference_range_2.yml',
+        ];
+
+        Fixtures::load($files, $managerMock, [ 'providers' => [ $this ] ]);
+    }
+
+    /**
+     * @test
+     * @expectedException \UnexpectedValueException
+     */
+    public function loadFixturesByReferenceSameFixtureNotFound()
+    {
+        $managerMock = $this->getDoctrineManagerMock(null);
+
+        $files = [
+            __DIR__ . '/support/fixtures/reference_range_3.yml',
+        ];
+
+        Fixtures::load($files, $managerMock, [ 'providers' => [ $this ] ]);
+    }
+
+    /**
+     * @test
+     */
+    public function loadFixturesByReference()
+    {
+        $managerMock = $this->getDoctrineManagerMock(7);
+
+        $managerMock->expects($this->exactly(2))
+            ->method('flush');
+
+        $files = [
+            __DIR__ . '/support/fixtures/reference_range_1.yml',
+            __DIR__ . '/support/fixtures/reference_range_2.yml',
+        ];
+        $objects = Fixtures::load($files, $managerMock, [ 'providers' => [ $this ] ]);
+
+        $this->assertCount(7, $objects);
+
+        $groupReference = $objects['group_1_user1'];
+        $this->assertInstanceOf(self::GROUP, $groupReference);
```

Comment:
besides testing that the constructed objects are of the right instance, it would be good to test the result value of `<current()>` for the `Group` entities